cppdock

```diff
@@ -155,16 +155,22 @@ def lock_item_json(item):
 def get_current_sha_with_comment(item):
     repo = item['name']
     tag = item['tag']
+    source = item.get('source')
 
     if 'revision' in item:
         return item['revision']
     else:
-        tag = normalize_branch_name(tag)
-        lines = subprocess.check_output(['git', 'ls-remote', normalize_repo_name(repo)]).splitlines()
-        for line in lines:
-            if line.endswith(tag):
-                return line[0:40]
-        raise ValueError("Unable to get SHA from remote git repository")
+        if source == 'docker':
+            line = subprocess.check_output(['docker', 'inspect', "--format='{{index .RepoDigests 0 }}'", repo + ":" + tag]).split(':')
+            return line[1][0:64]
+            raise ValueError("Unable to get SHA from docker image")
+        else:
+            tag = normalize_branch_name(tag)
+            lines = subprocess.check_output(['git', 'ls-remote', normalize_repo_name(repo)]).splitlines()
+            for line in lines:
+                if line.endswith(tag):
+                    return line[0:40]
+            raise ValueError("Unable to get SHA from remote git repository")
```

Comment:
Eventually this should be broken out into separate functions instead of nested if statements. just a stylistic nitpick